app/js/components/orderEntry/OrdersTable.jsx

```diff
@@ -1,30 +1,62 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import format from 'date-fns/format';
 import Accordion from './Accordion';
-import allOrdersMock from './allOrdersMock.json';
 import OrderHeader from './OrderHeader';
 import LabOrderDetails from './LabOrderDetails';
 import DrugOrderDetails from './DrugOrderDetails';
+import fetchOrder from '../../actions/fetchOrder';
 
-const OrdersTable = () => (
-  <React.Fragment>
-    {allOrdersMock.orders.map(order => (
-      <Accordion
-        title={<OrderHeader status={order.status} orderable={order.orderable} />}
-        key={order.id}>
-        {order.type === 'drugorder' ? (
-          <DrugOrderDetails
-            dosingInstructions={order.dosingInstructions}
-            dispense={order.dispense}
-            activeDates={order.activeDates}
-            orderer={order.orderer}
-          />
-        ) : (
-          <LabOrderDetails urgency={order.urgency} orderer={order.orderer} />
-        )}
-      </Accordion>
-    ))}
-  </React.Fragment>
-);
+export class OrdersTable extends PureComponent {
+  componentDidMount() {
```

Comment:
I am aware that there is a parent component for this, why not move the `componentLifeCycle` methods into that Parent Component.

Reply:
The `parent`, `grandParent` and `great-grandparents` of this component are all `presentational` components. 
If we decide to make the parent a `container` component, then we should also make it's grand-parent` a container component and pass down the data as a prop. 
I believe it's better to have one single container component down the line, than passing down props up to 3 layers to make it  `presentational`.
Also remember each component would be re-rendered, each time it's props is updated when it is being passed down.

Reply:
I think I agree with @jomadoye ... also, we'd want this OrdersTable component to be able to be used in multiple contexts, within different presentations/layouts, so doesn't it make sense for it to be responsible for fetching the data it plans to display? Or is there a different paradigm we should be following?